File: binds/java.v1/src/portalchell.java
import ch.qos.logback.classic.Logger;
import com.karalabe.iris.Connection;
import com.karalabe.iris.ServiceHandler;
import com.karalabe.iris.TopicHandler;
import com.karalabe.iris.exceptions.ClosedException;
import com.karalabe.iris.exceptions.InitializationException;
import org.slf4j.LoggerFactory;

import java.io.IOException;

public class PortalChellEntry implements ServiceHandler {
    public static void main(String args[]) throws IOException, InterruptedException, InitializationException, ClosedException {
        // Disable the logger
        final Logger logger = (Logger) LoggerFactory.getLogger(org.slf4j.Logger.ROOT_LOGGER_NAME);
        logger.detachAndStopAllAppenders();

// START OMIT
// Topic subscription handler processing inbound events
class Chell implements TopicHandler {
    @Override public void handleEvent(byte[] event) { // HLsub
        System.out.println(new String(event) + "\n");
    }
}
// Connect to the Iris network as Chell
try (Connection connection = new Connection(55555)) { // HLsub
    System.out.println("Tuning in to Aperture channels...");
    connection.subscribe("official", new Chell()); // HLsub

    Thread.sleep(60 * 1000);
}
// END OMIT
    }
}

File: binds/scala.v1/src/portalchell.scala
import ch.qos.logback.classic.Logger
import com.karalabe.iris.{Connection, TopicHandler}
import org.slf4j.LoggerFactory

object PortalChellEntry {
  // Disable the logger
  val logger: Logger = LoggerFactory.getLogger(org.slf4j.Logger.ROOT_LOGGER_NAME).asInstanceOf[Logger]
  logger.detachAndStopAllAppenders()

    def main(args: Array[String]) {
// START OMIT
// Topic subscription handler processing inbound events
class Chell extends TopicHandler {
    override def handleEvent(event: Array[Byte]) { // HLsub
        println(new String(event) + "\n")
    }
}
// Connect to the Iris network as Chell
val connection = new Connection(55555) // HLsub
try {
    println("Tuning in to Aperture channels...");
    connection.subscribe("official", new Chell) // HLsub

    Thread.sleep(60 * 1000)
} finally {
    connection.close()
}
// END OMIT
    }
}
